packages/connect/src/api/cardano/cardanoCertificate.ts

```diff
@@ -114,6 +115,16 @@ const validatePoolParameters = (poolParameters: CardanoPoolParameters) => {
     }
 };
 
+const validateDRep = (drep: CardanoDRep) => {
+    validateParams(drep, [{ name: 'type', type: 'number', required: true }]);
+
+    if (drep.type === PROTO.CardanoDRepType.KEY_HASH) {
+        validateParams(drep, [{ name: 'keyHash', type: 'string', required: true }]);
+    } else if (drep.type === PROTO.CardanoDRepType.SCRIPT_HASH) {
+        validateParams(drep, [{ name: 'scriptHash', type: 'string', required: true }]);
+    }
```

Comment:
What happens if the type is neither of these two options? Perhaps throw some error? (If we added KEY_PATH drep type one day, this function should not silently stop working. In general, we use switch statements for going through such enums, with `default` option resulting in some kind of assertion failure.)

Reply:
I think that in such case, the client should be warned by their type checker that the enum value doesn't exist. I agree with the switch-default approach in security-sensitive contexts. Here, it might help the client to identify an issue in some cases, but I don't think it's very different when the issue is caught by Connect here, vs. by Trezor a few moments later (?)